lib/cli-engine/config-array/config-array.js

```diff
@@ -139,13 +139,15 @@ function mergeWithoutOverwrite(target, source) {
     }
 
     for (const key of Object.keys(source)) {
-        if (key === "__proto__") {
+        if (!Object.prototype.hasOwnProperty.call(source, key)) {
```

Comment:
The key cames from `Object.keys(source)`, so I guess this is always true?

Reply:
I think it all practical cases this is probably true, however if may be possible for this to fail if source has some complex inheritance.

Given we've been bitten once before, it seems reasonable to be overly cautious here.

Reply:
Looking at the implementation of Object. 
keys this is probably safe to remove. I assume the proto check is left over from when the loop was some differently...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

Reply:
Hmm. I'm curious about your case. `Object.keys` is defined to enumerate only own properties: https://tc39.es/ecma262/#sec-object.keys

Reply:
Agreed.